lib/src/event.dart

```diff
@@ -582,7 +582,7 @@ class Event extends MatrixEvent {
   /// Gets the underlying mxc url of a thumbnail of a file event, or null if not present
   Uri? get thumbnailMxcUrl {
     final url = isThumbnailEncrypted
-        ? infoMap['thumbnail_file']['url']
+        ? (infoMap['thumbnail_file'] as Map)['url']
```

Comment:
Here the cast is even dangerous as it would break if this wasn't a Map. The JSON inside of the event content is fully user generated so an attacker could send `"thumbnail_file": 0.0` which would be a double and cause the whole app to crash here.

We would be safe by using tryGetMap here as well:
```dart
final url = isThumbnailEncrypted
  ? infoMap.tryGetMap<String, Object?>('thumbnail_file')?.tryGet<String>('url')
  : infoMap.tryGet<String>('thumbnail_url');
```